scripts/utils.sh

```diff
@@ -0,0 +1,384 @@
+#!/bin/bash
+#
+# Implements a package of utility functions.
+# Style Guide: https://google.github.io/styleguide/shellguide.html
+
+# Constants
+readonly DATE_FORMAT='%Y-%m-%d %H:%M:%S (%Z)'
+
+# Function's comment template
+#######################################
+# [Description of the function.]
+# Globals:
+#   [List of global variables used and modified, if any.]
+# Arguments:
+#   [Arguments taken, if any.]
+# Outputs:
+#   [Output to STDOUT or STDERR, if any.]
+# Returns:
+#   [Returned values other than the default exit status of the last command run.]
+#######################################
+
+#######################################
+# Add new commented entry in server hosts file.
+# Arguments:
+#   1) IP address to add
+#   2) Corresponding Fully Qualified Domain Name to add
+#   3) Corresponding entry comment to add
+# Outputs:
+#   Writes log to STDOUT.
+#######################################
+function utils::add_hosts_file_entry() {
+  local ip="$1"
+  local fqdn="$2"
+  local comment="$3"
+
+  readonly HOSTS_FILE_PATH="/etc/hosts"
+
+  utils::echo_action "Adding entry for ${fqdn} in ${HOSTS_FILE_PATH}..."
+  if ! grep -q "${fqdn}" "${HOSTS_FILE_PATH}"; then
+    printf "# ${comment}\n${ip} ${fqdn}\n" >> "${HOSTS_FILE_PATH}"
+    utils::echo_info "Done."
+  else
+    utils::echo_info "Skipped: ${HOSTS_FILE_PATH} already contains entry for ${fqdn}."
+  fi
+}
+
+#######################################
+# Echo a text string using the ACTION format.
+# Globals:
+#   DATE_FORMAT
+# Arguments:
+#   Message to echo
+# Outputs:
+#   Writes message to echo to STDOUT.
+#######################################
+function utils::echo_action() {
+  echo ""
+  echo "$(date +"$DATE_FORMAT") | ACTION - $1"
+}
+
+#######################################
+# Echo a text string using the ERROR format.
+# Globals:
+#   DATE_FORMAT
+# Arguments:
+#   Message to echo
+# Outputs:
+#   Writes message to echo to STDERR.
+#######################################
+function utils::echo_error() {
+  echo "$(date +"$DATE_FORMAT") | ERROR  - $1" >&2
+}
+
+#######################################
+# Echo a text string using the INFO format.
+# Globals:
+#   DATE_FORMAT
+# Arguments:
+#   Message to echo
+# Outputs:
+#   Writes message to echo to STDOUT.
```

Comment:
See previous comments, same thing for other occurrences

Reply:
Corrected.